tests/drivers/opamp/opamp_stm32/src/main.c

```diff
@@ -0,0 +1,379 @@
+			     "but it is 0x%x",
+			     opamp_dev->name, (uint32_t)LL_OPAMP_INPUT_INVERT_IO0_SEC,
+			     (uint32_t)LL_OPAMP_INPUT_INVERT_IO1_SEC, inm_sec);
+#endif /* st_inputs_mux_mode */
+
+		break;
+	case OPAMP_FUNCTIONAL_MODE_INVERTING:
+#if OPAMP_FILTERING_INPUT_PRESENT
+		zassert_equal(functional_mode, LL_OPAMP_MODE_PGA_IO0_IO1_BIAS,
+			      "%s: OPAMP shall be in LL_OPAMP_MODE_PGA_IO0_IO1_BIAS",
+			      opamp_dev->name);
+#else  /* OPAMP_FILTERING_INPUT_PRESENT */
+		zassert_equal(functional_mode, LL_OPAMP_MODE_PGA_IO0_BIAS,
+			      "%s: OPAMP shall be in LL_OPAMP_MODE_PGA_IO0_BIAS", opamp_dev->name);
+#endif /* OPAMP_FILTERING_INPUT_PRESENT */
+
+		/* VM_SEL = b10 in all PGA modes (OPAMPx_CSR register bits 6 to 5) */
+		zassert_true(IS_BIT_SET(OPAMP_LL_DEV->CSR, 6),
+			     "%s: OPAMP CSR VM_SEL (bit 6) shall be SET in "
+			     "inverting mode",
+			     opamp_dev->name);
+		zassert_false(IS_BIT_SET(OPAMP_LL_DEV->CSR, 5),
+			      "%s: OPAMP CSR VM_SEL (bit 5) shall be RESET in "
+			      "inverting mode",
+			      opamp_dev->name);
+
+#if DT_NODE_HAS_PROP(OPAMP_NODE, st_inputs_mux_mode)
+		/* The inverting mode is a sub-mode of PGA mode.
+		 * In this case VMS_SEL is defined by OPAMPx_TCMR register bit 0.
+		 * The value of VMS_SEL in OPAMPx_TCMR is therefore don't care here.
+		 *
+		 * From RM0440 Rev 9 pp. 809/2140:
+		 * When PGA (VM_SEL = “10”) or Follower mode (VM_SEL = “11”) is used:
+		 *	0: Resistor feedback output selected (PGA mode)
+		 *	1: VOUT selected as input minus (follower mode)
+		 */
+#endif /* st_inputs_mux_mode */
+		break;
+	case OPAMP_FUNCTIONAL_MODE_NON_INVERTING:
+#if OPAMP_FILTERING_INPUT_PRESENT
+		zassert_equal(functional_mode, LL_OPAMP_MODE_PGA_IO0,
+			      "%s: OPAMP shall be in LL_OPAMP_MODE_PGA_IO0", opamp_dev->name);
+#else /* OPAMP_FILTERING_INPUT_PRESENT */
+		zassert_equal(functional_mode, LL_OPAMP_MODE_PGA,
+			      "%s: OPAMP shall be in LL_OPAMP_MODE_PGA", opamp_dev->name);
+
+#endif /* OPAMP_FILTERING_INPUT_PRESENT */
+
+		/* VM_SEL = b10 in all PGA modes (OPAMPx_CSR register bits 6 to 5) */
+		zassert_true(IS_BIT_SET(OPAMP_LL_DEV->CSR, 6),
+			     "%s: OPAMP CSR VM_SEL (bit 6) shall be SET in "
+			     "inverting mode",
+			     opamp_dev->name);
+		zassert_false(IS_BIT_SET(OPAMP_LL_DEV->CSR, 5),
+			      "%s: OPAMP CSR VM_SEL (bit 5) shall be RESET in "
+			      "inverting mode",
+			      opamp_dev->name);
+		break;
+	case OPAMP_FUNCTIONAL_MODE_DIFFERENTIAL:
+		/* differential mode is not supported directly - device shall not be ready */
+		/* The device shall not be ready in differential mode */
+		zassert_false(device_is_ready(opamp_dev),
+			      "OPAMP device %s shall be NOT ready in differential mode",
+			      opamp_dev->name);
+		break;
+	default:
+		zassert_unreachable("Unsupported functional mode %d", dts_functional_mode);
+		break;
+	};
+
+	/* Gain shall have register reset value (0x0) after init */
+	const uint32_t gain = LL_OPAMP_GetPGAGain(OPAMP_LL_DEV);
```

Comment:
This local variable `gain` is named like the global `gain[]` array defined at line 38. Compiler seems happy but could you rename the global `gain[]` array into `programmable_gain[]`, for consistency?

Reply:
Renamed to programmable_gain